### PR TITLE
pinnwand, steck: update

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

New upstream releases

https://github.com/supakeen/pinnwand/releases/tag/v1.2.3
https://github.com/supakeen/steck/releases/tag/v0.7.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/pinnwand/default.nix [hunk 1]
```diff
@@ -37,19 +37,7 @@ in with python.pkgs; buildPythonApplication rec {
...
 
-  checkInputs = [ pytest ];
-
-  checkPhase = ''
-    pytest
-  '';
-
-  patches = [
-    # Use poetry-core instead of poetry. Fixed in 1.2.3
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/servers/pinnwand/default.nix:48]: Please remove fetchPatch from inputs.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108998` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>pinnwand</li>
<li>steck</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108998` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>pinnwand: log <a href="https://termbin.com/en6w">https://termbin.com/en6w</a></li>
  </ul>
</details>
<details>
<summary>
1 package built:
</summary>
<ul>
<li>steck</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
darwin patch:

```diff
diff --git a/pkgs/servers/pinnwand/default.nix b/pkgs/servers/pinnwand/default.nix
index 475e2c228de..807f241fa3e 100644
--- a/pkgs/servers/pinnwand/default.nix
+++ b/pkgs/servers/pinnwand/default.nix
@@ -39,6 +39,8 @@ in with python.pkgs; buildPythonApplication rec {

   checkInputs = [ pytestCheckHook ];

+  __darwinAllowLocalNetworking = true;
+
   passthru.tests = nixosTests.pinnwand;

   meta = with lib; {
   
```

**Comment by mweinelt** [comment]
@SuperSandro2000 test pinnwand

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108998` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>pinnwand</li>
<li>steck</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108998` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>pinnwand</li>
<li>steck</li>
</ul>
</details>

**Pull request merged**